vmas/simulator/dynamics/drone.py

```diff
@@ -0,0 +1,148 @@
+from typing import Union
+
+import torch
+from torch import Tensor
+import vmas.simulator.core
+import vmas.simulator.utils
+from vmas.simulator.dynamics.common import Dynamics
+
+
+class Drone(Dynamics):
+    def __init__(
+        self,
+        world: vmas.simulator.core.World,
+        I_xx: float = 8.1e-3,
+        I_yy: float = 8.1e-3,
+        I_zz: float = 14.2e-3,
+        integration: str = "rk4",
+    ):
+        super().__init__()
+
+        assert integration in (
+            "rk4",
+            "euler",
+        )
+        
+        self.integration = integration
+        self.I_xx = I_xx
+        self.I_yy = I_yy
+        self.I_zz = I_zz
+        self.world = world
+        self.g = 9.81
+        self.dt = world.dt
+        self.reset()
+        
+    def reset(self, index: Union[Tensor, int] = None):
+        if index is None:
+            # Drone state: phi(roll), theta (pitch), psi (yaw),
+            #              p (roll_rate), q (pitch_rate), r (yaw_rate),
+            #              x_dot (vel_x), y_dot (vel_y), z_dot (vel_z),
+            #              x (pos_x), y (pos_y), z (pos_z)
+            self.drone_state = torch.tensor(
+                [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], device=self.world.device
```

Comment:
these are all zeros now?

If so you can do `torch.zeros(self.world.batch_dim,12,device-self.world.device)`

Reply:
Yeah, because the initial height (the last element in the state) can be zero due to the lack of "ground" in VMAS. Also, it's easier to do the reset function. We allow the drone to fly at a negative altitude. We might need to think of a way to define the spatial constraint in the z-direction. (for need_reset() function or training in general).